pkg/cloudprovider/nodecontroller/nodecontroller.go

```diff
@@ -187,198 +144,6 @@ func (nc *NodeController) Run(period time.Duration, syncNodeList bool) {
 	}, nc.nodeMonitorPeriod)
 }
 
-// registerNodes registers the given list of nodes, it keeps retrying for `retryCount` times.
-func (nc *NodeController) registerNodes(nodes *api.NodeList, retryCount int, retryInterval time.Duration) error {
-	if len(nodes.Items) == 0 {
-		return nil
-	}
-
-	registered := util.NewStringSet()
-	nodes = nc.canonicalizeName(nodes)
-	for i := 0; i < retryCount; i++ {
-		for _, node := range nodes.Items {
-			if registered.Has(node.Name) {
-				continue
-			}
-			_, err := nc.kubeClient.Nodes().Create(&node)
-			if err == nil || apierrors.IsAlreadyExists(err) {
-				registered.Insert(node.Name)
-				glog.Infof("Registered node in registry: %s", node.Name)
-			} else {
-				glog.Errorf("Error registering node %s, retrying: %s", node.Name, err)
-			}
-			if registered.Len() == len(nodes.Items) {
-				glog.Infof("Successfully registered all nodes")
-				return nil
-			}
-		}
-		time.Sleep(retryInterval)
-	}
-	if registered.Len() != len(nodes.Items) {
-		return ErrRegistration
-	} else {
-		return nil
-	}
-}
-
-// reconcileExternalServices updates balancers for external services, so that they will match the nodes given.
-// Returns true if something went wrong and we should call reconcile again.
-func (nc *NodeController) reconcileExternalServices(nodes *api.NodeList) (shouldRetry bool) {
```

Comment:
I filed #6926 to move it there, but haven't done so yet. I figured @a-robinson would get it moved over before we had agreement on this PR and it was ready to be merged. Otherwise I'll move it before this is merged.